src/mpi/comm/comm_dup.c

```diff
@@ -160,8 +160,10 @@ int MPI_Comm_dup(MPI_Comm comm, MPI_Comm *newcomm)
     
     mpi_errno = MPIR_Comm_dup_impl(comm_ptr, &newcomm_ptr);
     if (mpi_errno) MPIR_ERR_POP(mpi_errno);
-
-    MPIR_OBJ_PUBLISH_HANDLE(*newcomm, newcomm_ptr->handle);
+    if (newcomm_ptr)
```

Comment:
This isn't quite right. If we are at this point in the code, we know that we have been given a valid pointer in `newcomm`. It should be safe to unconditionally publish the comm handle here. We only return `MPI_COMM_NULL` when the operation fails, which should be handled by the code after the `fn_fail` label.